### PR TITLE
feat: enter lat/long coordinates in map search

### LINKED ISSUE #370: Entering lat/long coordinates into search bar does not initiate valid search
## Description

Typing coordinates into the home screen search bar isn't working the way I would expect, i.e. dropping a temporary site marker right on that location. Seen on Android and iOS.

https://github.com/techmatters/terraso-mobile-client/assets/2730009/e7b3056d-0c38-46aa-950f-80a6b1904f2e

### PR DESCRIPTION
**Pull request by ajabeckett**
## Description
Adds coordinates check to home screen map search.

### Related Issues
Fixes #370

### Verification steps
Enter valid coordinates in the map search field and press enter on the keyboard. A temporary location popover should open at the correct coordinates.

### CODE REVIEW

File: dev-client/src/components/home/MapSearch.tsx [hunk 1]
```diff
@@ -82,24 +82,39 @@ export default function MapSearch({zoomTo, zoomToUser, toggleMapLayer}: Props) {
...
-      try {
-        const {suggestions: newSuggestions} =
-          await makeSuggestionsApiCall(queryText);
-        setSuggestions(newSuggestions);
-      } catch (e: any) {
-        if (e.name !== 'AbortError') {
-          throw e;
-        }
+    const coordRegex =
+      /^([-+]?[1-8]?\d(?:\.\d+)?),\s*([-+]?180(?:\.0+)?|[-+]?((1[0-7]\d)|([1-9]?\d))(?:\.\d+)?)$/;
```
Comment (paulschreiber) [inline dev-client/src/components/home/MapSearch.tsx:86]: question: what are you trying to match here? looks like a pair of decimal numbers or an integer or a decimal number? it's hard to follow. Is there a coord regex in one of the libraries we have that we can reuse?

File: dev-client/src/util.ts [hunk 2]
```diff
@@ -42,3 +42,17 @@ export const removeKeys = (a: any, b: any) => {
     }
   }
 };
+
+export const isValidCoordinates = (input: string) => {
+  const coordRegex =
+    /^([-+]?[1-8]?\d(?:\.\d+)?),\s*([-+]?180(?:\.0+)?|[-+]?((1[0-7]\d)|([1-9]?\d))(?:\.\d+)?)$/;
+
```
Comment (paulschreiber) [inline dev-client/src/util.ts:49]: issue: this is duplicating code in MapSearch.tsx.

File: dev-client/src/util.ts [hunk 3]
```diff
@@ -42,3 +42,17 @@ export const removeKeys = (a: any, b: any) => {
...
+  if (!coordRegex.test(input)) {
+    return false;
+  }
+
+  const [latitude, longitude] = input.split(',').map(Number);
+  return (
+    latitude >= -90 && latitude <= 90 && longitude >= -180 && longitude <= 180
+  );
```
Comment (paulschreiber) [inline dev-client/src/util.ts:57]: issue: We have `isValidLatitude` and `isValidLongitude` in gisUtils in the web client. Move those to shared client and reuse here.

File: dev-client/src/components/home/MapSearch.tsx [hunk 4]
```diff
@@ -82,24 +82,39 @@ export default function MapSearch({zoomTo, zoomToUser, toggleMapLayer}: Props) {
...
+
+    if (coordRegex.test(queryText)) {
```
Comment (paulschreiber) [inline dev-client/src/components/home/MapSearch.tsx:88]: issue: use isValidCoordinates() here.

**Review by paulschreiber — CHANGES_REQUESTED** [review]

File: dev-client/src/components/home/MapSearch.tsx (hunk 4 above)
Comment (ajabeckett) [reply dev-client/src/components/home/MapSearch.tsx:88]: Whoops, this was a stale file.

File: dev-client/src/util.ts (hunk 3 above)
Comment (ajabeckett) [reply dev-client/src/util.ts:57]: Will do, thanks!

**Review by paulschreiber — CHANGES_REQUESTED** [review]
Avoid duplicating work in MapSearch — logic should be in isValidCoordinates.

**Comment by ajabeckett** [comment]
Whoops, yes, that was an oversight, thanks for catching!

**Review by paulschreiber — APPROVED** [review]
